### PR TITLE
refactored edit user form for admin

### PR DESCRIPTION
**Pull request by o-rudych**
Now the form based on React Hook Form.

### CODE REVIEW

File: shared/src/validation-schemas/edit-user/edit-user.validation-schema.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as Yup from 'yup';
+
+const phoneReg = /^(?:\+38)?(0[5-9][0-9]\d{7})$/gm;
+
+const editUserSchema = Yup.object().shape({
+  name: Yup.string()
+    .required('Full name is required')
+    .max(150, 'Full name must not exceed 100 characters'),
```
Comment (stanislav-ostr) [inline shared/src/validation-schemas/edit-user/edit-user.validation-schema.ts]: Wrong description: 100 characters, but need 150.

File: frontend/src/components/administration/users/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    position: absolute;
+    top: 30px;
+    right: 30px;
+    cursor: pointer;
+  }
+
+  .mainTitle {
+    margin: 0 0 2.5rem;
+    font-weight: 800;
+    font-size: 18px;
```
Comment (stanislav-ostr) [inline frontend/src/components/administration/users/styles.module.scss]: Maybe you should use rem?

**Review by stanislav-ostr — CHANGES_REQUESTED** [review]

File: shared/src/validation-schemas/edit-user/edit-user.validation-schema.ts (hunk 1 above)
Comment (o-rudych) [reply shared/src/validation-schemas/edit-user/edit-user.validation-schema.ts]: Thanks, fixed.

File: frontend/src/components/administration/users/styles.module.scss (hunk 2 above)
Comment (o-rudych) [reply frontend/src/components/administration/users/styles.module.scss]: Makes sense, fixed.

**Comment by o-rudych** [comment]
Comments accepted. Fixed.
@stanislav-ostr @oleksandr-danylchenko

**Review by stanislav-ostr — DISMISSED** [review]
Thanks

File: frontend/src/components/administration/users/dialog-edit-user.tsx [hunk 3]
```diff
@@ -1,148 +1,155 @@
...
-    setEmail(user.email);
-    setPhone(user.phone);
-    setLocation(user.location);
-    setSex(user.sex);
-    setRole(user.role);
-  }, [user]);
+interface DialogEditUserProps {
+  user?: User;
+  onClose: () => void;
+  onSave: (user: User, newUserData: Partial<Omit<User, 'id'>>) => void;
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/administration/users/dialog-edit-user.tsx]: Usually, `onClosed` passes as the last prop

File: frontend/src/components/administration/users/dialog-edit-user.tsx [hunk 4]
```diff
@@ -1,148 +1,155 @@
...
-    setSex(event.target.value as unknown as UserSex);
-  };
+  useEffect(() => {
+    if (!user) return;
+    setValue('name', user.name);
+    setValue('email', user.email);
+    setValue('phone', user.phone);
+    setValue('location', user.location);
+    setValue('sex', user.sex);
+    setValue('role', user.role);
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/administration/users/dialog-edit-user.tsx]: This set of invocations can be simplified - https://stackoverflow.com/a/69536828/10963661

File: frontend/src/components/administration/users/users-list.tsx [hunk 5]
```diff
@@ -110,8 +110,8 @@ const UsersList: FC<UsersListProps> = ({ users }) => {
 
       <DialogEditUser
         user={selectedUser}
-        handleClose={handleClose}
-        handleSubmit={handleSubmit}
+        onClose={handleClose}
+        onSave={handleSubmit}
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/administration/users/users-list.tsx]: Usually `onClose` passed as the last one

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]

File: frontend/src/components/administration/users/users-list.tsx (hunk 5 above)
Comment (o-rudych) [reply frontend/src/components/administration/users/users-list.tsx]: OK, I got it. Done.

File: frontend/src/components/administration/users/dialog-edit-user.tsx (hunk 3 above)
Comment (o-rudych) [reply frontend/src/components/administration/users/dialog-edit-user.tsx]: Done

File: frontend/src/components/administration/users/dialog-edit-user.tsx (hunk 4 above)
Comment (o-rudych) [reply frontend/src/components/administration/users/dialog-edit-user.tsx]: Thanks. It's really a better way. Done.

**Review by oleksandr-danylchenko — APPROVED** [review]

**Pull request merged**